xmodule/tests/test_library_content.py

```diff
@@ -234,121 +281,147 @@ def test_validation_of_course_libraries(self):
+        self.lc_block.source_library_version = None
         result = self.lc_block.validate()
         assert not result
-        # Validation fails due to at least one warning/message
         assert result.summary
         assert StudioValidationMessage.WARNING == result.summary.type
         assert 'out of date' in result.summary.text
 
         # Now if we update the block, all validation should pass:
-        self.lc_block.refresh_children()
+        self._sync_lc_block_from_library()
         assert self.lc_block.validate()
 
+        # But updating the library will cause it to fail again as out-of-date:
+        self._add_problems_to_library()
+        result = self.lc_block.validate()
+        assert not result
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert 'out of date' in result.summary.text
+
+        # And a regular sync will not fix that:
+        self._sync_lc_block_from_library()
+        result = self.lc_block.validate()
+        assert not result
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert 'out of date' in result.summary.text
+
+        # But a upgrade_to_latest sync will:
+        self._sync_lc_block_from_library(upgrade_to_latest=True)
+        assert self.lc_block.validate()
+
+    def _assert_has_only_N_matching_problems(self, result, n):
+        assert result.summary
+        assert StudioValidationMessage.WARNING == result.summary.type
+        assert f'only {n} matching problem' in result.summary.text
+
     def test_validation_of_matching_blocks(self):
         """
         Test that the validation method of LibraryContent blocks can warn
         the user about problems with other settings (max_count and capa_type).
         """
+        # Ensure we're starting wtih clean validation
+        assert self.lc_block.validate()
+
         # Set max_count to higher value than exists in library
         self.lc_block.max_count = 50
-        # In the normal studio editing process, editor_saved() calls refresh_children at this point
-        self.lc_block.refresh_children()
         result = self.lc_block.validate()
         assert not result
-        # Validation fails due to at least one warning/message
-        assert result.summary
-        assert StudioValidationMessage.WARNING == result.summary.type
-        assert 'only 4 matching problems' in result.summary.text
+        self._assert_has_only_N_matching_problems(result, 4)
         assert len(self.lc_block.selected_children()) == 4
 
         # Add some capa problems so we can check problem type validation messages
+        self._add_problems_to_library()
+        self._sync_lc_block_from_library(upgrade_to_latest=True)
         self.lc_block.max_count = 1
-        self._create_capa_problems()
-        self.lc_block.refresh_children()
         assert self.lc_block.validate()
```

Comment:
I strengthened this test a bit: by setting max_count and then validating _without_ syncing from library, we confirm that reselection of a learner's blocks isn't contingent on a library sync.